compiler/modules/front/stmt.bal

```diff
@@ -1158,7 +1158,12 @@ function codeGenExprForCond(StmtContext cx, bir:BasicBlock bb, Environment env,
     }
     else if flags != 0 {
         bir:Register reg = cx.createTmpRegister(t:BOOLEAN);
-        bir:AssignInsn insn = { result: reg, operand: { value, semType: t:BOOLEAN }, pos: expr.startPos };
+        bir:EqualityInsn insn = {
+            op: "===",
```

Comment:
This is violating the rule that the intersection of the types of the operands of === must not be disjoint.

You need to use `===` and `!==` ops with the equal operands, as I suggested.

Reply:
fixed